### PR TITLE
docs(examples): add documentation for ampersand and css combinators

### PR DESCRIPTION
**Pull request by mtcrutch**
**What**:
Add documentation and an example of `&` usage with CSS combinators.

**Why**:
Usage wasn't super clear to the consumer before.

**How**:
Via [Twitter](https://twitter.com/_jayphelps/status/854484171938381824). Tweet tweet.

### CODE REVIEW

**Comment by mtcrutch** [comment]
Fixed!

Hit `enter` rather than `space` during the prompt 😁

File: examples/README.md [hunk 1]
```diff
@@ -51,6 +51,31 @@ const MyGrid = glamorous.div({
 })
 ```
 
+## Ampersands & CSS Combinators
+
+You can use an ampersand (`&`) as a reference to the parent selector. This approach can be useful for CSS combinators: adjacent sibling selectors (`+`), general sibling selectors (`~`) and child selectors (`>`).
```
Comment (kentcdodds) [inline examples/README.md]: Instead of "parent selector" I think we should describe it as "generated selector" or "current selector"

Comment (mtcrutch) [inline examples/README.md]: Good call; that's a lot cleaner. Updating with "generated selector."

**Comment by kentcdodds** [comment]
This is so great, thank you very much!

**Pull request merged**